code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -245,46 +272,321 @@
+	return ..()
+
+
+/datum/action/xeno_action/activable/hunter_mark/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/M = A
+
+	if(!isliving(M)) //Sanity
+		to_chat(X, "<span class='xenodanger'>We cannot mark this target!</span>")
+		fail_activate()
+
+	X.face_atom(M) //Face towards the target so we don't look silly
+
+	to_chat(X, "<span class='xenodanger'>We prepare to psychically mark [M.name] as our quarry.</span>")
+
+	if(!do_after(X, HUNTER_MARK_WINDUP, TRUE, target, BUSY_ICON_HOSTILE)) //Slight wind up
 		return fail_activate()
 
-	if(get_dist(X, victim) > 5)
-		to_chat(X, "<span class='warning'>They are too far for us to reach their minds!</spam>")
+	if(!X.line_of_sight(M)) //Need line of sight.
+		to_chat(X, "<span class='xenowarning'>We lost line of sight to the target!</span>")
+		return fail_activate()
+
+	X.hunter_mark_target = M //Set our target
+	RegisterSignal(X.hunter_mark_target, COMSIG_PARENT_PREQDELETED, .proc/unset_target) //For var clean up
+
+	to_chat(X, "<span class='xenodanger'>We psychically mark [M.name] as our quarry.</span>")
+	X.playsound_local(X, 'sound/effects/ghost.ogg', 25, 0, 1)
 
 	succeed_activate()
-	X.playsound_local(X.loc, 'sound/voice/4_xeno_roars.ogg', 30, TRUE)
-	to_chat(X, "<span class='notice'>We reach out into mind of the creature, infecting their thoughts...</span>")
-	victim.hallucination += 100
+
+	GLOB.round_statistics.hunter_marks++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "hunter_marks") //Statistics
+	add_cooldown()
+
+/datum/action/xeno_action/activable/hunter_mark/proc/unset_target()
+	var/mob/living/carbon/xenomorph/X = owner
+	UnregisterSignal(X.hunter_mark_target, COMSIG_PARENT_PREQDELETED)
+	X.hunter_mark_target = null
+
+// ***************************************
+// *********** Psychic Trace
+// ***************************************
+/datum/action/xeno_action/psychic_trace
+	name = "Psychic Trace"
+	action_icon_state = "toggle_queen_zoom"
+	mechanics_text = "Psychically ping the creature you marked, letting you know its direction, distance and location, and general condition."
+	plasma_cost = 1 //Token amount
+	keybind_signal = COMSIG_XENOABILITY_PSYCHIC_TRACE
+	cooldown_timer = HUNTER_PSYCHIC_TRACE_COOLDOWN
+
+/datum/action/xeno_action/psychic_trace/action_activate()
+
+	var/mob/living/carbon/xenomorph/hunter/X = owner
+	var/mob/living/M = X.hunter_mark_target
+
+	if(X.on_fire)
+		to_chat(X, "<span class='xenowarning'>We're too busy being on fire to trace!</span>")
+		return fail_activate()
+
+	if(!M)
+		to_chat(X, "<span class='xenowarning'>We have no target we can trace!</span>")
+		return fail_activate()
+
+	if(M.z != X.z)
+		to_chat(X, "<span class='xenowarning'>Our target is too far away, and is beyond our senses!</span>")
+		return fail_activate()
+
+	var/name = M.name
+	var/area = "[get_area(M)] (X: [M.x], Y: [M.y])"
+	var/distance = get_dist(X, M)
+	var/condition = calculate_mark_health()
```

Comment:
For these variables, you don't need the 'speed' of having a local var. Just use the actual calc in the message below.

(`var/condition` is excepted)